### PR TITLE
Fix chat issue

### PR DESCRIPTION
**Pull request by lyw0149**
1. Make goto chat page on activity page.
2. Delete onBack() when chatPage will unmount.

### CODE REVIEW

File: src/components/Chat/react-native-gifted-chat/src/MessageContainer.js [hunk 1]
```diff
@@ -75,10 +75,30 @@ export default class MessageContainer extends React.Component {
     const messagesData = this.prepareMessages(nextProps.messages);
     this.setState({
       dataSource: this.state.dataSource.cloneWithRows(messagesData.blob, messagesData.keys),
-      opponentInfoInfo: nextProps.opponentInfoInfo,
+      opponentInfo: nextProps.opponentInfo,
     });
   }
 
+  getWorkInfo(user) {
```
Comment (SarahhhC) [inline src/components/Chat/react-native-gifted-chat/src/MessageContainer.js]: Is this function for getting user's experience?
Now we replaced work as experience!

File: src/components/Chat/react-native-gifted-chat/src/MessageContainer.js [hunk 2]
```diff
@@ -75,10 +75,30 @@ export default class MessageContainer extends React.Component {
...
+      if (this.state.opponentInfo.work.length != 0) {
+        if (this.state.opponentInfo.work[0].position && this.state.opponentInfo.work[0].employer) {
+          workInfo = `${this.state.opponentInfo.work[0].position.name} at ` +
+            `${this.state.opponentInfo.work[0].employer.name}`;
+        } else {
+          if (this.state.opponentInfo.work[0].position) {
+            workInfo = this.state.opponentInfo.work[0].position.name;
+          } else if (this.state.opponentInfo.work[0].employer) {
+            workInfo = this.state.opponentInfo.work[0].employer.name;
+          }
```
Comment (SarahhhC) [inline src/components/Chat/react-native-gifted-chat/src/MessageContainer.js:94]: So you get only one recent work information from user, right?
Why this part is needed for message container?

**Comment by SarahhhC** [comment]
In `src/components/Activity/ConnectedRow.js` we had error that can't move to chatting page, so temporarily replaced 
`me: { userId: this.props.me._id },` => `me: { userId: this.props.dataSource._id },`
Please check this part too. Thanks 👍

File: src/components/Chat/react-native-gifted-chat/src/MessageContainer.js (hunk 2 above)
Comment (lyw0149) [reply src/components/Chat/react-native-gifted-chat/src/MessageContainer.js:94]: Because of MessageContainer Header

**Comment by SarahhhC** [comment]
Good Thanks @lyw0149 LGTM 👍

**Comment by MechanicKim** [comment]
I have confirmed that sufficient testing has been done. Thanks! LGTM

**Pull request merged**